### PR TITLE
Change SQL Server codenames to version names

### PR DESCRIPTION
**Pull request by Wraith2**
Take 2 of https://github.com/dotnet/SqlClient/pull/1334 from a clean main to fix weird things going on.
Each rename is an individual commit. I only did the main library not the tests and i didn't change the resource names. If they're important they can be done separately.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBuffer.cs [hunk 1]
```diff
@@ -499,7 +499,7 @@ internal string String
         }
 
         // use static list of format strings indexed by scale for perf
-        private static readonly string[] s_katmaiDateTimeOffsetFormatByScale = new string[] {
+        private static readonly string[] s_2008DateTimeOffsetFormatByScale = new string[] {
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBuffer.cs:502]: Could I ask you to have the version with `sql` prefix wherever it's meaningful like here?
```suggestion
        private static readonly string[] s_sql2008DateTimeOffsetFormatByScale = new string[] {
```

**Review by DavoudEshtehari — COMMENTED** [review]
Big thanks, It's too much clearer.

File: src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS/TDSVersion.cs [hunk 2]
```diff
@@ -12,17 +12,17 @@ namespace Microsoft.SqlServer.TDS
...
-        /// Katmai TDS version
+        /// 2008 (Katmai) TDS version
         /// </summary>
         public static Version SqlServer2008 = new Version(7, 3, 11, 3);
 
         /// <summary>
-        /// Denali TDS version
+        /// 2012 (Denali) TDS version
         /// </summary>
         public static Version SqlServer2010 = new Version(7, 4, 0, 4);
```
[the remark this answers is not in the document]
Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS/TDSVersion.cs]: Odd, the value was correct. Perhaps they started work on it in 2010. Changed.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBuffer.cs (hunk 1 above)
Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBuffer.cs]: Done.

**Review by DavoudEshtehari — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 3]
```diff
@@ -338,6 +338,7 @@
     </Compile>
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\SqlCommandBuilder.cs">
       <Link>Microsoft\Data\SqlClient\SqlCommandBuilder.cs</Link>
+      <SubType>Component</SubType>
     </Compile>
```
Comment (Kaur-Parminder) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:342]:
```suggestion
      <Link>Microsoft\Data\SqlClient\SqlCommandBuilder.cs</Link>
    </Compile>
```

Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:342]: The IDE is just going to keep putting it back until the PR that contains the change in the attribute on the class is merged. Can we just live with it until then?

Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:342]: I think you included changes in this [PR](https://github.com/dotnet/SqlClient/pull/1334#issuecomment-998339865) and closed it for pipeline failures. Those changes are missing here.

Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj]: Should be fixed and not come back now.

**Review by JRahnama — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Pull request merged**